pkg/sql/schemachanger/scplan/internal/opgen/opgen_composite_type_attribute.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package opgen
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scop"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+)
+
+func init() {
+	opRegistry.register((*scpb.CompositeTypeAttribute)(nil),
+		toPublic(
+			scpb.Status_ABSENT,
+			to(scpb.Status_PUBLIC,
+				emit(func(this *scpb.CompositeTypeAttribute) *scop.UpdateTypeBackReferencesInTypes {
+					if len(this.ClosedTypeIDs) == 0 {
+						return nil
+					}
+					var ids catalog.DescriptorIDSet
+					for _, id := range this.ClosedTypeIDs {
+						ids.Add(id)
+					}
+					return &scop.UpdateTypeBackReferencesInTypes{
+						TypeIDs:              ids.Ordered(),
```

Comment:
You should just plop `this.ClosedTypeIDs` in there after checking that it's not empty. It should be considered as part of `scdecomp`'s contract that the `TypeT` messages are well-formed. Same comment applies below.

Reply:
Nice, got it, Done.